### PR TITLE
Enable local backup by default on API level

### PR DESCRIPTION
**Pull request by moelsayed**
https://github.com/rancher/rancher/issues/17817

### CODE REVIEW

File: pkg/controllers/management/clusterprovisioner/driver.go [hunk 1]
```diff
@@ -12,18 +12,30 @@ import (
...
+	DefaultBackupRetention     = 6
+)
 
 func (p *Provisioner) driverCreate(cluster *v3.Cluster, spec v3.ClusterSpec) (api string, token string, cert string, err error) {
 	ctx, logger := clusterprovisioninglogger.NewLogger(p.Clusters, cluster, v3.ClusterConditionProvisioned)
 	defer logger.Close()
 
 	spec = cleanRKE(spec)
+	if cluster.Spec.RancherKubernetesEngineConfig != nil &&
+		cluster.Spec.RancherKubernetesEngineConfig.Services.Etcd.BackupConfig == nil { // no backup config
```
Comment (alena1108) [inline pkg/controllers/management/clusterprovisioner/driver.go]: we should enable it on the api layer, meaning that the field has to be set on the api layer, and persisted to etcd

File: pkg/api/store/cluster/cluster_store.go [hunk 2]
```diff
@@ -313,3 +320,19 @@ func validateNetworkFlag(data map[string]interface{}, create bool) error {
 
 	return nil
 }
+
+func enableLocalBackup(data map[string]interface{}) {
+	rkeConfig, ok := values.GetValue(data, "rancherKubernetesEngineConfig")
+
+	if ok && rkeConfig != nil {
+		backupConfig := values.GetValueN(data, "rancherKubernetesEngineConfig", "services", "etcd", "backupConfig")
+		if backupConfig == nil {
```
Comment (alena1108) [inline pkg/api/store/cluster/cluster_store.go]: should we also check the legacy snapshot value? And if not nil, and == true, don't configure the backup config

File: pkg/api/store/cluster/cluster_store.go [hunk 3]
```diff
@@ -313,3 +320,19 @@ func validateNetworkFlag(data map[string]interface{}, create bool) error {
...
+			backupConfig = &v3.BackupConfig{
```
Comment (alena1108) [inline pkg/api/store/cluster/cluster_store.go]: just wanted to confirm if the backupConfig value doesn't need an explicit conversion into map, and gets saved as intended on the rke config

Comment (moelsayed) [reply pkg/api/store/cluster/cluster_store.go]: it's stored correctly in the config.

**Comment by alena1108** [comment]
LGTM

**Pull request merged**